mage_ai/orchestration/db/models/schedules.py

```diff
@@ -120,6 +121,19 @@ def last_pipeline_run_status(self) -> str:
             return None
         return sorted(self.pipeline_runs, key=lambda x: x.created_at)[-1].status
 
+    @property
+    def trigger_as_code_exists(self) -> bool:
```

Comment:
This method is not necessarily needed.
In `add_or_update_trigger_for_pipeline_and_persist` or `remove_trigger` method, we can check whether the trigger exists. If it exists, we update or remove the trigger.